test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/PrngPrecompileSuite.java

```diff
@@ -0,0 +1,230 @@
+ * limitations under the License.
+ */
+package com.hedera.services.bdd.suites.contract.precompile;
+
+import static com.hedera.services.bdd.spec.HapiApiSpec.defaultHapiSpec;
+import static com.hedera.services.bdd.spec.assertions.ContractFnResultAsserts.isRandomResult;
+import static com.hedera.services.bdd.spec.assertions.ContractFnResultAsserts.resultWith;
+import static com.hedera.services.bdd.spec.assertions.TransactionRecordAsserts.recordWith;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getTxnRecord;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCall;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.uploadInitCode;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.sourcing;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CONTRACT_REVERT_EXECUTED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PrngPrecompileSuite extends HapiApiSuite {
+    private static final Logger log = LogManager.getLogger(PrngPrecompileSuite.class);
+    private static final long GAS_TO_OFFER = 400_000L;
+    private static final String THE_GRACEFULLY_FAILING_PRNG_CONTRACT = "GracefullyFailingPrng";
+    private static final String bob = "bob";
+
+    public static void main(String... args) {
+        new PrngPrecompileSuite().runSuiteSync();
+    }
+
+    @Override
+    public boolean canRunConcurrent() {
+        return false;
+    }
+
+    @Override
+    public List<HapiApiSpec> getSpecsInSuite() {
+        return allOf(positiveSpecs(), negativeSpecs());
+    }
+
+    List<HapiApiSpec> negativeSpecs() {
+        return List.of(
+                functionCallWithLessThanFourBytesFailsGracefully(),
+                nonSupportedAbiCallGracefullyFails(),
+                invalidLargeInputFails(),
+                emptyInputCallFails());
+    }
+
+    List<HapiApiSpec> positiveSpecs() {
+        return List.of(prngPrecompileHappyPathWorks());
+    }
+
+    private HapiApiSpec emptyInputCallFails() {
+        final var prng = THE_GRACEFULLY_FAILING_PRNG_CONTRACT;
+        final var emptyInputCall = "emptyInputCall";
+        return defaultHapiSpec("emptyInputCallFails")
+                .given(cryptoCreate(bob), uploadInitCode(prng), contractCreate(prng))
+                .when(
+                        sourcing(
+                                () ->
+                                        contractCall(prng, "performEmptyInputCall")
```

Comment:
`withExplicitParams` with a zero length array is what I think is needed.

Reply:
Fixed to use `withExplicitParams`